### PR TITLE
feat(meta): support schema change in actor graph builder

### PR DESCRIPTION
**Pull request by BugenZhao**
Signed-off-by: Bugen Zhao <i@bugenzhao.com>

I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This PR extends the new actor graph builder introduced in #7659 for replacing table plan, and adds the procedure of preparing the `TableFragments`. Note that the RPC with the compute nodes is not implemented so this is not utilized by the frontend yet.

Building an actor graph for schema change can be similar and symmetric to the MV on MV.
- For MV on MV, we have existing upstream `Materialize` fragments that give the requirements to the distribution of current `Chain`. We'll generate new dispatchers for the upstream `Materialize` in the result.
- For replacing table plan for schema change, we (may) have existing downstream `Chain` fragments that give the requirements to the distribution of current `Materialize`. We'll generate updates for the merger for the downstream `Chain` in the result.

Related:
- #7908

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: proto/stream_plan.proto [hunk 1]
```diff
@@ -44,6 +44,10 @@ message UpdateMutation {
     // Merge executor can be uniquely identified by a combination of actor id and upstream fragment id.
     uint32 actor_id = 1;
     uint32 upstream_fragment_id = 2;
+    // - For scaling, this is always `None`.
+    // - For plan change, the upstream fragment will be changed to a new one, and this will be `Some`.
+    //   In this case, all the upstream actors should be removed and replaced by the `new` ones.
+    optional uint32 new_upstream_fragment_id = 5;
```
Comment (BugenZhao) [inline proto/stream_plan.proto:50]: We may reuse the `Update` barrier of scaling for schema change.

File: src/meta/src/rpc/service/ddl_service.rs [hunk 2]
```diff
@@ -581,18 +587,12 @@ where
 
         // 2. Get the env for streaming jobs.
         let env = StreamEnvironment::from_protobuf(fragment_graph.get_env().unwrap());
-        let default_parallelism = if let Some(Parallelism { parallelism }) =
```
Comment (BugenZhao) [inline src/meta/src/rpc/service/ddl_service.rs:584]: We put this into `FragmentGraph::new`.

**Review by chenzl25 — APPROVED** [review]
LGTM!  It's interesting to think of schema changes and mv on mv as dual problems in some sense. I like this idea.

**Review by yezizp2012 — APPROVED** [review]
LSTM!

**Pull request merged**